File: app/src/main/java/com/josegonzalez/musicsearchsample/view/search/SearchActivity.java
package com.josegonzalez.musicsearchsample.view.search;

import android.os.Bundle;
import android.view.KeyEvent;

import com.google.android.material.floatingactionbutton.FloatingActionButton;
import com.google.android.material.textfield.TextInputEditText;
import com.josegonzalez.musicsearchsample.R;
import com.josegonzalez.musicsearchsample.presenter.SearchPresenter;
import com.josegonzalez.musicsearchsample.view.BaseActivity;

import static com.josegonzalez.musicsearchsample.view.itemList.ItemListActivity.ITEM_LIST_DEEP_LINK;
import static com.josegonzalez.musicsearchsample.view.itemList.ItemListActivity.WORD_TO_SEARCH_KEY;

public class SearchActivity extends BaseActivity implements SearchView {

    public static final String HOME_DEEP_LINK = HOST + "home";

    private TextInputEditText editText;

    private SearchPresenter searchPresenter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.search_activity);

        searchPresenter = new SearchPresenter(this);

        init();
    }

    private void init() {
        editText = findViewById(R.id.wordToSearch);
        editText.setOnKeyListener((v, keyCode, event) -> {
            if ((event.getAction() == KeyEvent.ACTION_DOWN) && (keyCode == KeyEvent.KEYCODE_ENTER)) {
                onSearchButtonPressed();
                return true;
            }
            return false;
        });

        FloatingActionButton floatingActionButton = findViewById(R.id.fab);
        floatingActionButton.setOnClickListener(v -> onSearchButtonPressed());
    }

    public void onSearchButtonPressed() {
        searchPresenter.onSearch(editText.getText().toString());
    }

    public void goToMusicList(String wordToSearch) {
        final Bundle bundle = new Bundle();
        bundle.putString(WORD_TO_SEARCH_KEY, wordToSearch);
        startActivityDeepLink(ITEM_LIST_DEEP_LINK, bundle);
    }

}

File: app/src/main/java/com/josegonzalez/musicsearchsample/features/search/SearchActivity.kt
package com.josegonzalez.musicsearchsample.features.search

import android.os.Bundle
import android.view.KeyEvent
import android.view.View
import com.google.android.material.floatingactionbutton.FloatingActionButton
import com.google.android.material.textfield.TextInputEditText
import com.josegonzalez.musicsearchsample.R
import com.josegonzalez.musicsearchsample.common.BaseActivity
import com.josegonzalez.musicsearchsample.common.ITEM_LIST_DEEP_LINK
import com.josegonzalez.musicsearchsample.features.list.ItemListActivity

class SearchActivity : BaseActivity(), SearchView {

    private var editText: TextInputEditText? = null
    private var searchPresenter: SearchPresenter? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.search_activity)
        searchPresenter = SearchPresenter(this)
        init()
    }

    private fun init() {
        editText = findViewById(R.id.wordToSearch)
        if (editText != null) {
            editText!!.setOnKeyListener(View.OnKeyListener { _: View?, keyCode: Int, event: KeyEvent ->
                if (event.action == KeyEvent.ACTION_DOWN && keyCode == KeyEvent.KEYCODE_ENTER) {
                    onSearchButtonPressed()
                    return@OnKeyListener true
                }
                false
            })
        }
        val floatingActionButton = findViewById<FloatingActionButton>(R.id.fab)
        floatingActionButton.setOnClickListener { onSearchButtonPressed() }
    }

    override fun onSearchButtonPressed() {
        searchPresenter!!.onSearch(editText!!.text.toString())
    }

    override fun goToMusicList(wordToSearch: String?) {
        val bundle = Bundle()
        bundle.putString(ItemListActivity.WORD_TO_SEARCH_KEY, wordToSearch)
        startActivityDeepLink(ITEM_LIST_DEEP_LINK, bundle)
    }

}